onnxscript/rewriter/pattern_test.py

```diff
@@ -780,6 +780,159 @@ def test_model1(x: FLOAT[16, 32], y: FLOAT[16, 32]) -> FLOAT[16, 32]:
+                # Get the numpy array from const_value
+                numpy_array = value.const_value.numpy()
+                
+                # Check if it represents a single value and is positive
+                if numpy_array.size != 1:
+                    return False
+                    
+                return float(numpy_array.item()) > 0
+            
+            # If const_value is None, check if this value is produced by a Constant node
+            if hasattr(value, 'producer'):
+                producer = value.producer()  # Call the method
+                
+                if producer is not None and producer.op_type == "Constant":
+                    # Check for value_float attribute
+                    if 'value_float' in producer.attributes:
+                        float_val = producer.attributes['value_float'].value
+                        return float_val > 0
+                    # Check for value_int attribute
+                    elif 'value_int' in producer.attributes:
+                        int_val = producer.attributes['value_int'].value
+                        return int_val > 0
+            
+            return False
+        
+        # Create a pattern with value checker using callable directly
+        def add_pattern(op, x, y):
+            # Use callable as input to create ValuePattern with checker
+            checked_x = is_positive_constant  # This should be converted to ValuePattern with check
+            return op.Add(checked_x, y)
+        
+        # Create the pattern
+        rule_pattern = pattern.Pattern(add_pattern)
+        
+        # Create a model with several calls to Add:
+        # - one with first parameter non-constant
+        # - one with first parameter a positive constant 
+        # - one with first parameter a negative constant
+        model_proto = onnx.parser.parse_model(
+            """
+            <ir_version: 7, opset_import: [ "" : 17]>
+            agraph (float[N] x, float[N] y) => (float[N] z1, float[N] z2, float[N] z3)
+            {
+                pos_const = Constant <value_float = 2.5> ()
+                neg_const = Constant <value_float = -1.5> ()
+                z1 = Add(x, y)           # non-constant first parameter
+                z2 = Add(pos_const, y)   # positive constant first parameter  
+                z3 = Add(neg_const, y)   # negative constant first parameter
+            }
+            """
+        )
+        model = ir.serde.deserialize_model(model_proto)
+        
+        # Find the Add nodes in the model
+        nodes = list(model.graph.all_nodes())
+        add_nodes = [node for node in nodes if node.op_type == "Add"]
+        self.assertEqual(len(add_nodes), 3)
+        
+        # Test case 1: Non-constant first parameter - should not match
+        match_result = rule_pattern.match(model, model.graph, add_nodes[0])
+        self.assertIsNone(match_result)
```

Comment:
Check that `bool(match_result)` is False or True (as appropriate) in all the tests, instead of checking for None.